app/util/error_handler.rb

```diff
@@ -16,7 +19,10 @@ module ErrorHandler
         :invalid_email => 400,
         :invalid_password => 400,
         :user_exists => 400,
```

Comment:
I feel these errors shoud all be just invalid request. The details could be added args passed to render_error. Which I think I added in my commits...